codec/src/main/java/io/netty/handler/codec/DelimitedLengthFieldBasedFrameDecoder.java

```diff
@@ -0,0 +1,178 @@
+     */
+    @Override
+    protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
+        if (discardingTooLongFrame) {
+            long bytesToDiscard = this.bytesToDiscard;
+            int localBytesToDiscard = (int) Math.min(bytesToDiscard, in.readableBytes());
+            in.skipBytes(localBytesToDiscard);
+            bytesToDiscard -= localBytesToDiscard;
+            this.bytesToDiscard = bytesToDiscard;
+
+            failIfNecessary(false);
+
+            // if exception was not thrown, reader has been advanced by readable bytes already, so return null to
+            // continue processing too long frame
+            return null;
+        }
+
+        if (consumingLength) {
+            int delimIndex = indexWithin(in, delimiter);
+            if (delimIndex < 0) {
+                return null;
+            }
+
+            frameLength = extractFrameLength(in, delimIndex);
+
+            if (frameLength < 0) {
+                throw new CorruptedFrameException("negative pre-adjustment length field: " + frameLength);
+            }
+
+            // this will not overflow because lengthAdjustment is an int
+            frameLength += lengthAdjustment;
+
+            //consume length field and delimiter bytes
+            in.skipBytes(delimIndex + delimiter.capacity());
+
+            //consume delimiter bytes
+            consumingLength = false;
+        }
+
+        if (frameLength > maxFrameLength) {
+            long discard = frameLength - in.readableBytes();
+            tooLongFrameLength = frameLength;
+
+            if (discard < 0) {
+                // buffer contains more bytes then the frameLength so we can discard all now
+                // will not overflow because frameLength must be less than in.readableBytes, an int
+                in.skipBytes((int) frameLength);
+            } else {
+                // Enter the discard mode and discard everything received so far.
+                discardingTooLongFrame = true;
+                bytesToDiscard = discard;
+                in.skipBytes(in.readableBytes());
+            }
+            failIfNecessary(true);
+            return null;
+        }
+
+        // never overflows because it's less than maxFrameLength
+        int frameLengthInt = (int) frameLength;
+        if (in.readableBytes() < frameLengthInt) {
+            // need more bytes available to read actual frame
+            return null;
+        }
+
+        // extract frame
+        int readerIndex = in.readerIndex();
+        ByteBuf frame = extractFrame(in, readerIndex, frameLengthInt);
+        in.readerIndex(readerIndex + frameLengthInt);
+
+        // the frame is now entirely present, reset state vars
+        consumingLength = true;
+        frameLength = 0;
+
+        return frame;
+    }
+
+    private int extractFrameLength(ByteBuf in, int delimIndex) {
+        if (delimIndex > maxPossibleLengthInBytes) {
```

Comment:
This does not take into account possible whitespaces which are assumed by the `trimLengthString` param.